ignite/distributed/comp_models/native.py

```diff
@@ -499,3 +491,84 @@ def _expand_hostlist(nodelist: str) -> List[str]:
                 result_hostlist += final_hostlist
 
         return result_hostlist
+
+    def _setup_ddp_vars_from_slurm_env(environ: Dict[str, str]) -> Dict[str, Union[str, int]]:
+        """Method to setup DDP env vars required by PyTorch from SLURM env
+        """
+        # 1) Tools like enroot can have hooks to translate slurm env vars to RANK, LOCAL_RANK, WORLD_SIZE etc
+        # See https://github.com/NVIDIA/enroot/blob/v3.1.0/conf/hooks/extra/50-slurm-pytorch.sh
+        # 2) User can use torch.distributed.launch tool to schedule on N local GPUs using 1 node, 1 task by SLURM
+        # To cover case 1), let's ensure that defined RANK == SLURM_PROCID, LOCAL_RANK == SLURM_LOCALID,
+        #   WORLD_SIZE == SLURM_NTASKS. We will use defined MASTER_ADDR and MASTER_PORT instead of defining
+        #   them by our means
+        # To cover case 2), let's check that defined RANK >= SLURM_PROCID, LOCAL_RANK >= SLURM_LOCALID,
```

Comment:
If I understand what is done, the idea is to ensure that in such a case, the user didn't use `srun` as a mistake 
```
srun python -m torch.distributed.launch ...
```
Therefore, every process should have a rank, local rank and world size greater or equal to what is defined by slurm.

Is it correct ?

Reply:
The case 2 is to cover use-case like : 
```
srun -N1 -n1 -G8 python -m torch.distributed.launch\
        --nproc_per_node=8 --nnodes=1 --node_rank=0 \
        --master_addr="localhost" --master_port=1234 \
        main.py
```

Reply:
case 2)

`RANK >= SLURM_PROCID`, `LOCAL_RANK >= SLURM_LOCALID` means that one process is spawn on the node by `srun` 
but 
`WORLD_SIZE >= SLURM_NTASKS` sounds weird. `SLURM_NTASKS` is the max number of tasks checked by slurm. If `WORLD_SIZE` is greater to that value, the scheduler should kill the process, because more than allocated ressources for the job are used.

What do you think ?

Reply:
Otherwise, it works on my side.

Reply:
The example above `srun -N1 -n1` allocates `SLURM_NTASKS=1`, but launcher creates ws=8. That's why `WORLD_SIZE >= SLURM_NTASKS`. Am I missing something ?

Reply:
Yes but the slurm scheduler can kill the job if more ressources than allocated are used. I suppose it depends on how the scheduler is configured. Imagine you schedule a job defining 4 tasks and in fact you use 8, it could be a big issue in production. Anyway, I think that does not really matter. This is a user constraint, we can't handle.